src/main/java/com/questhelper/quests/sleepinggiants/SleepingGiants.java

```diff
@@ -0,0 +1,300 @@
+
+	@Override
+	public Map<Integer, QuestStep> loadSteps()
+	{
+		setupZones();
+		setupItemRequirements();
+		setupConditions();
+		setupSteps();
+
+		Map<Integer, QuestStep> steps = new HashMap<>();
+
+		ConditionalStep startQuest = new ConditionalStep(this, strikeHillGiant);
+		startQuest.addStep(new Conditions(LogicType.NOR, onDesertPlateau), goToDesertPlateau);
+		startQuest.addStep(new Conditions(LogicType.NOR, hammer), takeHammer);
+		steps.put(0, startQuest);
+
+		ConditionalStep speakToKovacOnDesertPlateau = new ConditionalStep(this, speakToKovac);
+		speakToKovacOnDesertPlateau.addStep(new Conditions(LogicType.NOR, onDesertPlateau), goToDesertPlateau);
+		startQuest.addStep(new Conditions(LogicType.NOR, hammer), takeHammer);
+		steps.put(5, speakToKovacOnDesertPlateau);
+
+		ConditionalStep fixMachinery = new ConditionalStep(this, speakToKovacAfterRepairs);
+		fixMachinery.addStep(new Conditions(LogicType.NOR, inGiantsFoundry), enterFoundry);
+		fixMachinery.addStep(new Conditions(LogicType.NOR, polishingStoneFixed), fixPolishingStone);
+		fixMachinery.addStep(new Conditions(LogicType.NOR, grindstoneFixed), fixGrindstone);
+		fixMachinery.addStep(new Conditions(LogicType.NOR, hammerFixed), fixHammer);
+		steps.put(10, fixMachinery);
+		steps.put(15, fixMachinery);
+
+		ConditionalStep speakToKovacContinueAfterRepairs = new ConditionalStep(this, speakToKovacAfterRepairs);
+		speakToKovacContinueAfterRepairs.addStep(new Conditions(LogicType.NOR, inGiantsFoundry), enterFoundry);
+		steps.put(20, speakToKovacContinueAfterRepairs);
+
+		// TODO: This needs validation, as it's a very complex varbit-base step. It is very possible that the varbits are not checked properly.
+		ConditionalStep speakToKovacForCommission = new ConditionalStep(this, speakToKovacContinue);
+		speakToKovacForCommission.addStep(new Conditions(LogicType.NOR, inGiantsFoundry), enterFoundry);
+		speakToKovacForCommission.addStep(new Conditions(commissionReceived), searchCrate);
+		speakToKovacForCommission.addStep(new Conditions(crateSearched), fillCrucible);
+		speakToKovacForCommission.addStep(new Conditions(selectingMould, noForteSelected, forteTabOpen), selectForte);
+		speakToKovacForCommission.addStep(new Conditions(selectingMould, noForteSelected), selectForteTab);
+		speakToKovacForCommission.addStep(new Conditions(selectingMould, noBladeSelected, bladeTabOpen), selectBlade);
+		speakToKovacForCommission.addStep(new Conditions(selectingMould, noBladeSelected), selectBladesTab);
+		speakToKovacForCommission.addStep(new Conditions(selectingMould, noTipSelected, tipTabOpen), selectTip);
+		speakToKovacForCommission.addStep(new Conditions(selectingMould, noTipSelected), selectBlade);
+		speakToKovacForCommission.addStep(new Conditions(selectingMould), setMould);
+		speakToKovacForCommission.addStep(new Conditions(shouldSetMould), interactWithMould);
```

Comment:
There needs to be an additional step before this to speak to Kovac again before setting the mould.